### ISSUE TITLE
undefined symbol: _Z8bind_guiv

### ISSUE THREAD

**Issue by 0312birdzhang** [davisking/dlib#53]
I use <code>python_examples/compile_dlib_python_module.bat</code> build dlib.so,and copy dlib.so to 
<code>/usr/lib/python2.7/site-packages/</code>
Logs:
```
[mersdk@SailfishSDK python_examples]$ python
Python 2.7.6 (default, May 09 2013, 12:34:56) 
[GCC 4.8.3 20140106 (Mer 4.8.3-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import dlib
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: ./dlib.so: undefined symbol: _Z8bind_guiv
```
The same issue like http://sourceforge.net/p/dclib/discussion/442518/thread/c05888de/

**Comment by davisking** [comment]
Try using the 'pip install dlib' command.  Does that work?

**Comment by 0312birdzhang** [comment]
@davisking 
I had installed python2.7-devel,python3,boost
```
[root@Jolla nemo]# python3 -m pip install dlib
You are using pip version 6.0.8, however version 7.1.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Collecting dlib
  Downloading dlib-18.17.100.zip (4.4MB)
    100% |################################| 4.4MB 18kB/s 
    package init file 'dist/dlib/__init__.py' not found (or not a regular file)
Installing collected packages: dlib
  Running setup.py install for dlib
    Detected Python architecture: 32bit
    Configuring cmake ...
    -- The C compiler identification is GNU 4.8.3
    -- The CXX compiler identification is GNU 4.8.3
    -- Check for working C compiler: /usr/bin/cc
    -- Check for working C compiler: /usr/bin/cc -- works
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Check for working CXX compiler: /usr/bin/c++
    -- Check for working CXX compiler: /usr/bin/c++ -- works
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Could NOT find Boost
    -- Boost version: 1.51.0
    -- Found the following Boost libraries:
    --   python
    CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
      Could NOT find PythonLibs (missing: PYTHON_LIBRARIES PYTHON_INCLUDE_DIRS)
      (Required is at least version "3.4")
    Call Stack (most recent call first):
      /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:315 (_FPHSA_FAILURE_MESSAGE)
      /usr/share/cmake/Modules/FindPythonLibs.cmake:186 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
      /tmp/pip-build-nnavp_5m/dlib/dlib/add_python_module:64 (FIND_PACKAGE)
      CMakeLists.txt:5 (include)
    -- Configuring incomplete, errors occurred!
    error: cmake configuration failed!
    Complete output from command /usr/bin/python3 -c "import setuptools, tokenize;__file__='/tmp/pip-build-nnavp_5m/dlib/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-czyu8uuf-record/install-record.txt --single-version-externally-managed --compile:
    running install
    
    running build
    
    Detected Python architecture: 32bit
    
    Configuring cmake ...
    
    -- The C compiler identification is GNU 4.8.3
    
    -- The CXX compiler identification is GNU 4.8.3
    
    -- Check for working C compiler: /usr/bin/cc
    
    -- Check for working C compiler: /usr/bin/cc -- works
    
    -- Detecting C compiler ABI info
    
    -- Detecting C compiler ABI info - done
    
    -- Check for working CXX compiler: /usr/bin/c++
    
    -- Check for working CXX compiler: /usr/bin/c++ -- works
    
    -- Detecting CXX compiler ABI info
    
    -- Detecting CXX compiler ABI info - done
    
    -- Could NOT find Boost
    
    -- Boost version: 1.51.0
    
    -- Found the following Boost libraries:
    
    --   python
    
    CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
    
      Could NOT find PythonLibs (missing: PYTHON_LIBRARIES PYTHON_INCLUDE_DIRS)
    
      (Required is at least version "3.4")
    
    Call Stack (most recent call first):
    
      /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:315 (_FPHSA_FAILURE_MESSAGE)
    
      /usr/share/cmake/Modules/FindPythonLibs.cmake:186 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
    
      /tmp/pip-build-nnavp_5m/dlib/dlib/add_python_module:64 (FIND_PACKAGE)
    
      CMakeLists.txt:5 (include)
    
    -- Configuring incomplete, errors occurred!
    
    error: cmake configuration failed!
    
    ----------------------------------------
    Command "/usr/bin/python3 -c "import setuptools, tokenize;__file__='/tmp/pip-build-nnavp_5m/dlib/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-czyu8uuf-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-nnavp_5m/dlib

```

**Comment by davisking** [comment]
You seem to be missing the python development files.  You need to install
those too to be able to compile python modules.

**Comment by 0312birdzhang** [comment]
@davisking I had installed python3-dev,python2.7-dev and some development packages.
I can build out dlib.so .

**Comment by 0312birdzhang** [comment]
Use dlib-0.18.17.tar.gz and disable <code>SSE4</code> fixed this issue on sailfish os.
Though has other issue,i think it's time to close this.

**Comment by nickrobinson** [comment]
@0312birdzhang Were you ever able to get this completely working. I am seeing the same issue at the moment with 18.17.100.

**Comment by nickrobinson** [comment]
Just in case someone runs across this in the future. Diving a bit deeper it appeared that the bind_gui symbol from tools/python/src/dlib.cpp was getting included in the generated dlib.so even when I compiled with the DLIB_NO_GUI_SUPPORT flag. Since I had this flag disabled anyways I ended up commenting out the bind_gui definition and execution in tools/python/src/dlib.cpp and then recompiling.

I verified that the symbol was no longer present in my dlib.so by doing: ```nm dlib.so | grep bind_gui```

At this point everything started to work like normal and I was able to import dlib in my python project.
